src/js/game/builder.js

```diff
@@ -1,58 +1,52 @@
 import Celestial from "./ingameObjects/celestial";
 import Sprite from "../engine/sprite";
 import SpaceShip from "./ingameObjects/spaceship";
-import Explode from "./ingameObjects/explode";
 import ImageLoader from "../engine/imageLoader";
 import gameImages from "./gameImages";
+import PhysicalBody from "../engine/physicalbody";
+import CellestialImplementation from "../engine/implementation/cellestial.implementation";
+import EntityImplementation from "../engine/implementation/entity.implementation";
+import PhysicalBodyImplementation from "../engine/implementation/physicalBody.implementation";
+import SpriteImplementation from "../engine/implementation/sprite.implementation";
 
 
 export default class Builder {
-    #planets = {};
-    constructor(planets) {
-        this.#planets = planets;
-    }
 
-    buildAssets(){
+    buildAssets() {
         const loader = new ImageLoader();
         loader.setAssetList(gameImages);
         loader.loadImages();
         return loader;
     }
 
 
-
-    build(presets) {
-        let planet = this.#planets.spawnEmpty(presets.type);
-        planet.x = presets.x;
-        planet.y = presets.y;
-        planet.m = presets.m;
-        planet.vx = presets.vx;
-        planet.vy = presets.vy;
-        planet.m = presets.m;
-        planet.r = presets.r;
-        let sprite = new Sprite();
-        sprite.load(presets.image).then(image => {
-            sprite.setImage(image);
-            sprite.setLoaded();
-        });
-        let celestial = new Celestial(sprite,planet);
-        celestial.setFrameCount(presets.frameCount);
-        celestial.setFrameSize(presets.frameSize);
-        return celestial;
-
+    build(presets, engine) {
+        const buildData = presets.cellestial;
+        const cellestialInstance = new CellestialImplementation(
+            new EntityImplementation(
+                {
+                    label: buildData.entity.label,
+                    UUID: crypto.randomUUID()
+                }),
+            new PhysicalBodyImplementation(buildData.physicalBody),
+            new SpriteImplementation(buildData.sprite)
+        );
+        engine.eventHandlingSystem.addListener("onCollided", () => celestial.markForRemoval = true);
```

Comment:
nit-picking: Not sure if line 34 suits well build method. Seems like this can be handled outised this method.

Reply:
yes :D - you are right - once again - this has to be moved out of here